tuf/api/metadata.py

```diff
@@ -305,9 +304,7 @@ def sign(
         Raises:
             tuf.api.serialization.SerializationError:
                 'signed' cannot be serialized.
```

Comment:
This is not raised anymore. Can you move the serialize call out of the try block?

Reply:
My comment from above is nonsense. What I wanted to say is, if we except all below when calling `sign(serialize...)` and re-raise as `UnsignedMetadataError`, then `SerializationError` won't be raised anymore.

So either we take the call to serialize out of the try block below or we update the docstring

Reply:
I decided to move the serialization outside the try block so that the serialization error will propagate as I don't think it's good to hide it with `UnsignedMetadataError` exception.

Reply:
wow I completely missed this. Good solution I think.